source/extensions/filters/http/ext_proc/headers.cc

```diff
@@ -0,0 +1,45 @@
+#include "extensions/filters/http/ext_proc/headers.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExternalProcessing {
+
+using Http::LowerCaseString;
+
+void buildHttpHeaders(const Http::HeaderMap& headers_in,
+                      envoy::config::core::v3::HeaderMap* headers_out) {
+  headers_in.iterate([headers_out](const Http::HeaderEntry& e) -> Http::HeaderMap::Iterate {
+    auto new_header = headers_out->add_headers();
+    new_header->set_key(std::string(e.key().getStringView()));
+    new_header->set_value(std::string(e.value().getStringView()));
+    return Http::HeaderMap::Iterate::Continue;
+  });
+}
```

Comment:
Let see what else develops here -- so far ext_proc has a similar loop but it's not quite the same because they are filtering out a specific header.

Reply:
Actually, the other function in here, which turns an array of "HeaderValueOption" protobufs to changes in a header map, is worse -- we have two implementations in Envoy already, but the one in the "router" assumes that it can make string_views of header values, which won't work here, and the one in ext_authz is more abstract because it also tries to support HTTP responses. So I now have a third and I'm not sure how much it's worth trying to refactor all three.